File: FrogRiverOne.cs
using System;
using System.Linq;
using System.Collections.Generic;

class Solution {
    public int solution(int X, int[] A) {
        if (A.Length == 1) 
            return X == A.First() ? 0 : -1;
        
        if (!A.Contains(X))
            return -1;
       
        if (A.Length < X)
            return -1;
 
       HashSet<int> positions = new HashSet<int>();
     
       for(int index = 0; index < A.Length; index++)
       {
            positions.Add(A[index]);  
            
            if (positions.Count == X)
                return index;
       }
           
       return -1; 
    }
}

File: frog_river_one.py
def solution(X, A):
    if len(A) == 1:
        return 0 if X == A[0] else -1;
        
    if not X in A:
        return -1;
        
    if len(A) < X:
        return -1;
        
    positions = set()
    
    for index in range(0, len(A)):
        positions.add(A[index])
        
        if len(positions) == X:
            return index;
    
    return -1;
        